packages/format-library/src/underline/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { toggleFormat } from '@wordpress/rich-text';
+import { RichTextToolbarButton, RichTextShortcut } from '@wordpress/editor';
+
+const name = 'core/underline';
+
+export const underline = {
+	name,
+	title: __( 'Underline' ),
+	tagName: 'u',
+	className: null,
+	edit( { isActive, value, onChange } ) {
+		const onToggle = () => onChange( toggleFormat( value, { type: name } ) );
+
+		return (
+			<Fragment>
+				<RichTextShortcut
+					type="primary"
+					character="u"
+					onUse={ onToggle }
+				/>
+				<RichTextToolbarButton
```

Comment:
Does this add a toolbar button or is this component just required to make the shortcut work?

If this isn't visible it's worth noting that in a comment; it's kinda confusing. (If it is visible, didn't we decided awhile back not to include underlines in the toolbar? According to #7162 the lack of a button was intentional.)

Reply:
No the button is not required. It should be removed.

Reply:
Hi @tofumatt and @iseulde thank you for your reviews.
I am adding the button here button the button is not one of the default controls (they are `formattingControls: [ 'bold', 'italic', 'link', 'strikethrough' ]`) and no blocks currently use the button.
Adding the button here just opens the possibility for a block to have an underline button if the block wants to by changing the formattingControls prop. If we don't add the button here even if the block wants to show the underline the button will not render.

Reply:
Ah, so it's mainly for third-party blocks that might want to extend `RichText` and show that button.

That makes sense, but then a comment mentioning why it's there/not rendered by default (basically what you said here) would be really helpful in explaining why it's there 😄

Reply:
Given that in the past the button was explicitly removed and just the keyboard shortcode was allowed, other blocks should follow the same approach and we don't expect blocks to have the underline button so I end up removing the button as suggested by @iseulde and @tofumatt.
If a block really wants the button it is still possible using a custom format.